File: app/src/main/java/com/mapmyindia/sdk/demo/java/adapter/GeoAnalyticsAdapter.java
package com.mapmyindia.sdk.demo.java.adapter;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CompoundButton;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.switchmaterial.SwitchMaterial;
import com.mapmyindia.sdk.demo.R;
import com.mapmyindia.sdk.demo.java.model.GeoAnalyticsModel;

import java.util.List;

public class GeoAnalyticsAdapter extends RecyclerView.Adapter<GeoAnalyticsAdapter.ViewHolder> {

    private List<GeoAnalyticsModel> geoAnalyticsModels;
    private OnLayerSelected onLayerSelected;

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_geoanalytics_adapter, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, @SuppressLint("RecyclerView") int position) {
        GeoAnalyticsModel model = geoAnalyticsModels.get(position);
        holder.checkBox.setText(model.getType().getName());
        holder.checkBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if(onLayerSelected != null) {
                    onLayerSelected.onLayerSelected(geoAnalyticsModels.get(position), isChecked);
                }
            }
        });
    }

    public void setGeoAnalyticsModels(List<GeoAnalyticsModel> geoAnalyticsModels) {
        this.geoAnalyticsModels = geoAnalyticsModels;
        notifyDataSetChanged();
    }

    public void setOnLayerSelected(OnLayerSelected onLayerSelected) {
        this.onLayerSelected = onLayerSelected;
    }

    @Override
    public int getItemCount() {
        if(geoAnalyticsModels == null) {
            return 0;
        }
        return geoAnalyticsModels.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private SwitchMaterial checkBox;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            checkBox = itemView.findViewById(R.id.cb_geoanalytics);
        }
    }
    public interface OnLayerSelected {
        void onLayerSelected(GeoAnalyticsModel geoAnalyticsModel, boolean isChecked);
    }
}

File: app/src/main/java/com/mapmyindia/sdk/demo/kotlin/adapter/GeoAnalyticsAdapter.kt
package com.mapmyindia.sdk.demo.kotlin.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.CompoundButton
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.switchmaterial.SwitchMaterial
import com.mapmyindia.sdk.demo.R
import com.mapmyindia.sdk.demo.kotlin.model.GeoAnalyticsModel


/**
 * Created by Saksham on 27-09-2021
 */

class GeoAnalyticsAdapter: RecyclerView.Adapter<GeoAnalyticsAdapter.ViewHolder>() {

    private var onLayerSelected: OnLayerSelected? = null
    private var geoAnalyticsModels: List<GeoAnalyticsModel>? = null;


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.layout_geoanalytics_adapter, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val model = geoAnalyticsModels!![position]
        holder.checkBox.text = model.type.getName()
        holder.checkBox.setOnCheckedChangeListener { _, isChecked ->
            onLayerSelected?.onLayerSelected(
                model,
                isChecked
            )
        }
    }

    override fun getItemCount(): Int {
        return geoAnalyticsModels?.size?:0
    }

    fun setGeoAnalyticsModels(geoAnalyticsModels: List<GeoAnalyticsModel>) {
        this.geoAnalyticsModels = geoAnalyticsModels
        notifyDataSetChanged()
    }

    fun setOnLayerSelected(onLayerSelected: OnLayerSelected?) {
        this.onLayerSelected = onLayerSelected
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val checkBox: SwitchMaterial = itemView.findViewById(R.id.cb_geoanalytics)

    }

    interface OnLayerSelected {
        fun onLayerSelected(geoAnalyticsModel: GeoAnalyticsModel, isSelected: Boolean)
    }

}